tests/components/generic_thermostat/test_climate.py

```diff
@@ -271,6 +273,44 @@ def setup_comp_2(hass):
     assert temp == state.attributes.get("current_temperature")
 
 
+async def test_sensor_unknown(hass):
+    """Test when target sensor is Unknown."""
+    hass.states.async_set("sensor.unknown", STATE_UNKNOWN)
+    assert await async_setup_component(
+        hass,
+        "climate",
+        {
+            "climate": {
+                "platform": "generic_thermostat",
+                "name": "unknown",
+                "heater": ENT_SWITCH,
+                "target_sensor": "sensor.unknown",
+            }
+        },
+    )
+    state = hass.states.get("climate.unknown")
+    assert state.attributes.get("current_temperature") is None
+
+
+async def test_sensor_unavailable(hass):
+    """Test when target sensor is Unavailable."""
+    hass.states.async_set("sensor.unavailable", STATE_UNAVAILABLE)
+    assert await async_setup_component(
+        hass,
+        "climate",
+        {
+            "climate": {
+                "platform": "generic_thermostat",
+                "name": "unavailable",
+                "heater": ENT_SWITCH,
+                "target_sensor": "sensor.unavailable",
+            }
+        },
+    )
+    state = hass.states.get("climate.unavailable")
+    assert state.attributes.get("current_temperature") is None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5222933)